File: app/src/main/java/com/worldventures/dreamtrips/modules/trips/model/filter/CachedTripFilters.java
package com.worldventures.dreamtrips.modules.trips.model.filter;

import com.esotericsoftware.kryo.DefaultSerializer;
import com.esotericsoftware.kryo.serializers.CompatibleFieldSerializer;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

@DefaultSerializer(CompatibleFieldSerializer.class)
public class CachedTripFilters implements Serializable {

   private List<ActivityModel> activities = new ArrayList<>();
   private List<RegionModel> regions = new ArrayList<>();

   public CachedTripFilters() {
      //do nothing
   }

   public CachedTripFilters(List<RegionModel> regions, List<ActivityModel> activities) {
      this.activities = activities;
      this.regions = regions;
   }

   public List<ActivityModel> getActivities() {
      return activities;
   }

   public List<RegionModel> getRegions() {
      return regions;
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/modules/trips/model/filter/CachedTripFilters.kt
package com.worldventures.dreamtrips.modules.trips.model.filter

import com.esotericsoftware.kryo.DefaultSerializer
import com.esotericsoftware.kryo.serializers.CompatibleFieldSerializer
import java.io.Serializable

@DefaultSerializer(CompatibleFieldSerializer::class)
data class CachedTripFilters(val activities: List<ActivityModel> = emptyList(),
                             val regions: List<RegionModel> = emptyList()) : Serializable
